arch/arm/core/aarch32/cortex_m/mpu/arm_mpu.c

```diff
@@ -15,6 +15,16 @@
 #include <logging/log.h>
 LOG_MODULE_DECLARE(mpu);
 
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+#define MPU_NODEID DT_INST(0, arm_armv7m_mpu)
```

Comment:
@galak there is a problem here - ARMv6-M should be using the armv7-M mpu

Reply:
> > @ioannisg is there a header I can put this that is visible to cmsis.h and irq.h?
> 
> I think cmsis.h includes soc.h, which should be including both devicetree.h and vendor-specific headers (which should be including cmsis headers). What exactly would you like to include here?

My issue is where to define the NUM_IRQ_PRIO_BITS so there is visibility to both cmsis.h and irq.h.  I can introduce a new header

Reply:
> @galak there is a problem here - ARMv6-M should be using the armv7-M mpu

how so, what armv6-m have that dts node?

Reply:
V6-m SoCs may have MPU , but the mpu architecture is the v7-M MPU architecture. We dont' have an arm_armv6m_mpu node defined (maybe we should??)

Reply:
We should, I added a binding for it in the PR.  But updating the v6m dts files I feel like is outside the scope of this PR.